File: implementation/scipamato/core/web/src/test/java/ch/difty/scipamato/core/config/ScipamatoPropertiesTest.java
package ch.difty.scipamato.core.config;

import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.api.Test;

import ch.difty.scipamato.core.logic.parsing.AuthorParserStrategy;

class ScipamatoPropertiesTest {

    private final ScipamatoProperties sp = new ScipamatoProperties();

    @Test
    void brand_hasDefaultValue() {
        assertThat(sp.getBrand()).isEqualTo("SciPaMaTo-Core");
    }

    @Test
    void defaultLocalization_hasDefaultEnglish() {
        assertThat(sp.getDefaultLocalization()).isEqualTo("en");
    }

    @Test
    void pubmedBaseUrl_hasDefaultValue() {
        assertThat(sp.getPubmedBaseUrl()).isEqualTo("https://www.ncbi.nlm.nih.gov/pubmed/");
    }

    @Test
    void authorParser_isDefault() {
        assertThat(sp.getAuthorParser()).isEqualTo("DEFAULT");
    }

    @Test
    void authorParserStrategy_isDefault() {
        assertThat(sp.getAuthorParserStrategy()).isEqualTo(AuthorParserStrategy.PUBMED);
    }

    @Test
    void paperMinimumToBeRecycled_hasDefaultValue() {
        assertThat(sp.getPaperNumberMinimumToBeRecycled()).isEqualTo(0);
    }

    @Test
    void dbSchema_hasDefaultValuePublic() {
        assertThat(sp.getDbSchema()).isEqualTo("public");
    }

    @Test
    void gettingRedirectPort_hasNoDefaultValue() {
        assertThat(sp.getRedirectFromPort()).isNull();
    }

    @Test
    void gettingMultiSelectBoxActionBoxWithMoreEntriesThan_hasDefaultValue() {
        assertThat(sp.getMultiSelectBoxActionBoxWithMoreEntriesThan()).isEqualTo(4);
    }

    @Test
    void gettingPubmedApiKey_hasNoDefaultValue() {
        assertThat(sp.getPubmedApiKey()).isNull();
    }
}

File: core/core-web/src/test/kotlin/ch/difty/scipamato/core/config/ScipamatoPropertiesTest.kt
package ch.difty.scipamato.core.config

import ch.difty.scipamato.core.logic.exporting.RisExporterStrategy
import ch.difty.scipamato.core.logic.parsing.AuthorParserStrategy
import org.amshove.kluent.shouldBeEqualTo
import org.amshove.kluent.shouldBeNull
import org.junit.jupiter.api.Test

internal class ScipamatoPropertiesTest {

    private val sp = ScipamatoProperties()

    @Test
    fun brand_hasDefaultValue() {
        sp.brand shouldBeEqualTo "SciPaMaTo-Core"
    }

    @Test
    fun defaultLocalization_hasDefaultEnglish() {
        sp.defaultLocalization shouldBeEqualTo "en"
    }

    @Test
    fun pubmedBaseUrl_hasDefaultValue() {
        sp.pubmedBaseUrl shouldBeEqualTo "https://www.ncbi.nlm.nih.gov/pubmed/"
    }

    @Test
    fun cmsUrlSearchPage_hasNoDefaultValue() {
        sp.cmsUrlSearchPage.shouldBeNull()
    }

    @Test
    fun authorParser_isDefault() {
        sp.authorParser shouldBeEqualTo "DEFAULT"
    }

    @Test
    fun authorParserStrategy_isDefault() {
        sp.authorParserStrategy shouldBeEqualTo AuthorParserStrategy.PUBMED
    }

    @Test
    fun risExporterStrategy_isDefault() {
        sp.risExporterStrategy shouldBeEqualTo RisExporterStrategy.DEFAULT
    }

    @Test
    fun paperMinimumToBeRecycled_hasDefaultValue() {
        sp.paperNumberMinimumToBeRecycled shouldBeEqualTo 0
    }

    @Test
    fun dbSchema_hasDefaultValuePublic() {
        sp.dbSchema shouldBeEqualTo "public"
    }

    @Test
    fun gettingRedirectPort_hasNoDefaultValue() {
        sp.redirectFromPort.shouldBeNull()
    }

    @Test
    fun gettingMultiSelectBoxActionBoxWithMoreEntriesThan_hasDefaultValue() {
        sp.multiSelectBoxActionBoxWithMoreEntriesThan shouldBeEqualTo 4
    }

    @Test
    fun gettingPubmedApiKey_hasNoDefaultValue() {
        sp.pubmedApiKey.shouldBeNull()
    }
}
